### PR TITLE
Process withdrawals from Snapshot

### PR DESCRIPTION
**Pull request by zktony**
## Describe your changes
Process withdrawals from Snapshot.

## Issue ticket number and link
Will close https://github.com/Polkadex-Substrate/orderbook/issues/755

### CODE REVIEW

File: clients/orderbook/src/worker.rs [hunk 1]
```diff
@@ -440,6 +440,7 @@ where
 					}
 
 					let withdrawals = self.pending_withdrawals.clone();
+					//TODO: Extra logic to handle it
```
Comment (serhii-temchenko) [inline clients/orderbook/src/worker.rs]: Which logic should be here before removing `pending_withdrawals`? If it should not be implemented in this PR - please create issue and add inline comment with reference to that issue.

File: pallets/ocex/src/lib.rs [hunk 2]
```diff
@@ -1038,6 +1040,26 @@ pub mod pallet {
 				}
 				// Update the snapshot nonce and move the summary to snapshots storage
 				<SnapshotNonce<T>>::put(working_summary.snapshot_id);
+				//TODO: Handle Withdrawal
```
Comment (serhii-temchenko) [inline pallets/ocex/src/lib.rs]: Isn't this logic added on the next lines? Please elaborate a bit on this todo.

File: pallets/ocex/src/lib.rs [hunk 3]
```diff
@@ -1193,6 +1215,52 @@ pub mod pallet {
...
+						.try_insert(recipient_account, pending_withdrawals)
+						.map_err(|_| Error::<T>::WithdrawalBoundOverflow)?;
+				}
+			}
+			Ok(withdrawal_map)
+		}
+
+		// fn convert_account_id32_to_account_id<T, Z: Convert<AccountId32,
+		// T::AccountId>>(account_id32: AccountId32) -> T::AccountId { 	account_id32.into()
+		// }
```
Comment (serhii-temchenko) [inline pallets/ocex/src/lib.rs]: Please remove

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

**Review by felixfaisal — APPROVED** [review]
LGTM

File: pallets/ocex/src/lib.rs [hunk 4]
```diff
@@ -1193,6 +1214,48 @@ pub mod pallet {
 			Self::deposit_event(Event::WithdrawFromOrderbook(user, asset, amount));
 			Ok(())
 		}
+
+		fn create_withdrawal_tree(
+			pending_withdrawals: Vec<Withdrawal<AccountId32>>,
+		) -> Result<WithdrawalsMap<T>, sp_runtime::DispatchError> {
```
Comment (nakul1010) [inline pallets/ocex/src/lib.rs:1220]: plz, remove `sp_runtime`  `DispatchError` already imported above or it should be handled in this [issue](https://github.com/Polkadex-Substrate/Polkadex/issues/684)

File: pallets/ocex/src/lib.rs [hunk 5]
```diff
@@ -1193,6 +1214,48 @@ pub mod pallet {
 			Self::deposit_event(Event::WithdrawFromOrderbook(user, asset, amount));
 			Ok(())
 		}
+
+		fn create_withdrawal_tree(
```
Comment (nakul1010) [inline pallets/ocex/src/lib.rs:1218]: Can we create an issue for this,  to refactor the snapshot summary to `BtreeMap` on the client side or are we agreeing to use this in the future as well?

**Review by nakul1010 — APPROVED** [review]

File: pallets/ocex/src/lib.rs (hunk 5 above)
Comment (zktony) [reply pallets/ocex/src/lib.rs:1218]: done

**Pull request merged**